File: Cc.Core/Models/Region.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class Region
    {
        public Region()
        {
            InverseParent = new HashSet<Region>();
        }

        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public string? InsertUser { get; set; }
        public DateTime UpdateDate { get; set; }
        public string? UpdateUser { get; set; }
        public long CompanyId { get; set; }
        public long? ParentId { get; set; }
        public string? Code { get; set; }
        public string? Description { get; set; }

        public virtual Company Company { get; set; } = null!;
        public virtual Region? Parent { get; set; }
        public virtual ICollection<Region> InverseParent { get; set; }
    }
}

File: Cc.Ui/src/models/region.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('region')
export class Region extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    companyId: number;
    @field()
    parentId: number;
    @field()
    code: string;
    @field()
    description: string;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Region {

        let item = new Region();
        return item;
    }
}

@resource('region')
export class RegionSearchModel extends Entity implements Pagination {
    
    @field()
    companyId?: number;
    @field()
    parentId?: number;
    @field()
    code?: string;
    @field()
    description?: string;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}